### PR TITLE
Rename 'basic' fixture 'hello-cloud'

### PR DESCRIPTION
**Pull request by KnVerey**
Fixes https://github.com/Shopify/kubernetes-deploy/issues/50

### CODE REVIEW

**Comment by kirs** [comment]
There's a failing test; is it the one we detected as flaky?

**Comment by KnVerey** [comment]
> There's a failing test; is it the one we detected as flaky?

It's not the only one that has failed on CI, but looking over a number of builds, it does seem to be the primary offender, yes.

**Comment by KnVerey** [comment]
Hm, that failure suggests it manages to get into a state where there's only one pod running, and it is the one from the second of the two deploys in the test. If the pods from the first deploy were killed in the expected way, they would not be cleaned up... my guess is that something before the container start (e.g. image pull) is taking too long, so the pod doesn't have a chance to reach its timeout before the second deploy starts. I'll try replacing the sleep with a poll for at least one failed pod.

**Comment by KnVerey** [comment]
Push a fix attempt in the last commit, and it passed on this run anyway 😅 
Quick 👀  on that?

File: test/integration/kubernetes_deploy_test.rb [hunk 1]
```diff
@@ -106,13 +106,19 @@ def test_dead_pods_in_old_replicaset_are_ignored
...
     deploy_fixtures("hello-cloud", subset: ["web.yml.erb", "configmap-data.yml"])
     pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=hello-cloud")
-    running_pods, not_running_pods = pods.partition { |pod| pod.status.phase == "Running" }
-    assert_equal 1, running_pods.size
-    assert not_running_pods.size >= 1
+    running_pod_count = pods.count { |pod| pod.status.phase == "Running" }
+    final_failed_pod_count = pods.count { |pod| pod.status.phase == "Failed" }
+
+    assert_equal 1, running_pod_count
+    assert final_failed_pod_count >= initial_failed_pod_count # failed pods not cleaned up
```
Comment (kirs) [inline test/integration/kubernetes_deploy_test.rb]: Next time you can use `assert_operator 5, :<=, 4` that generates a slightly better failure message

File: test/integration/kubernetes_deploy_test.rb [hunk 2]
```diff
@@ -106,13 +106,19 @@ def test_dead_pods_in_old_replicaset_are_ignored
...
+      pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=hello-cloud")
+      initial_failed_pod_count = pods.count { |pod| pod.status.phase == "Failed" }
+    end
 
     deploy_fixtures("hello-cloud", subset: ["web.yml.erb", "configmap-data.yml"])
     pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=hello-cloud")
-    running_pods, not_running_pods = pods.partition { |pod| pod.status.phase == "Running" }
-    assert_equal 1, running_pods.size
-    assert not_running_pods.size >= 1
+    running_pod_count = pods.count { |pod| pod.status.phase == "Running" }
```
Comment (kirs) [inline test/integration/kubernetes_deploy_test.rb]: 👍 for using `count`

**Pull request merged**